dspace-api/src/main/java/org/dspace/core/ManagedHibernateDBConnection.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.core;
+
+import java.sql.SQLException;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+/**
+ * Implementing class for the MANAGED Context state for {@link HibernateDBConnection}
+ */
+public class ManagedHibernateDBConnection extends HibernateDBConnection {
+
+    @Autowired(required = true)
+    @Qualifier("managedSessionFactory")
+    private SessionFactory sessionFactory;
+
+    private Session session;
+    private Transaction transaction;
+
+    public SessionFactory getSessionFactory() {
+        return sessionFactory;
+    }
+
+    @Override
+    public Session getSession() {
+        if (session == null) {
+            this.session = getSessionFactory().openSession();
```

Comment:
Note: This call to `openSession()` will always establish a *new* Hibernate Session, rather than looking at whether there's a current Hibernate Session in this thread.